pkg/cfn/builder/cluster.go

```diff
@@ -79,7 +79,7 @@ func (c *ClusterResourceSet) addResourcesForControlPlane(version string) {
 	}
 
 	c.newResource("ControlPlane", &gfn.AWSEKSCluster{
-		Name:               gfn.NewString(c.spec.ClusterName),
+		Name:               gfn.NewString(c.spec.ID.Name),
```

Comment:
`.ID` should be `.Locator`, as struct name is `ClusterLocator`, because it's not an ID or ARN in the sense of AWS API.